.github/workflows/publish.yaml

```diff
@@ -0,0 +1,162 @@
+        run: |
+          helm template test helm/ --debug
+
+      - name: "Package Helm chart"
+        run: |
+          helm package helm/ --destination .
+
+      - name: "Upload chart package as artifact"
+        uses: "actions/upload-artifact@v4"
+        with:
+          name: "helm-chart"
+          path: "*.tgz"
+
+  #######################################################################
+  # Publish Helm Chart to GHCR
+  #######################################################################
+  publish-helm:
+    # Only push on tag or push to main.
+    if: github.ref_type == 'tag' || github.ref == 'refs/heads/main'
+    name: "Push Helm chart to GHCR"
+    runs-on: "ubuntu-latest"
+    needs: ["build-helm"]
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: "Check out repository"
+        uses: "actions/checkout@v5"
+
+      - name: "Set up Helm"
+        uses: "azure/setup-helm@v4"
+        with:
+          version: "latest"
+
+      - name: "Download chart package"
+        uses: "actions/download-artifact@v4"
+        with:
+          name: "helm-chart"
+
+      - name: "Set CHART_VERSION environment variable"
+        run: |
+          CHART_VERSION=$(grep '^version:' helm/Chart.yaml | awk '{print $2}')
+          echo "CHART_VERSION=$CHART_VERSION" >> $GITHUB_ENV
+
+      - name: "GHCR Login"
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | helm registry login ${{ env.REGISTRY }} -u ${{ github.actor }} --password-stdin
+
+      - name: "Push Helm chart to GHCR"
+        run: |
+          CHART_PACKAGE="${{ env.CHART_NAME }}-${{ env.CHART_VERSION }}.tgz"
+          
+          # Push to GHCR
+          helm push "${CHART_PACKAGE}" oci://${{ env.REGISTRY }}/${{ github.repository_owner }}/charts
+          
+          echo "✅ Successfully pushed ${CHART_PACKAGE} to ${{ env.REGISTRY }}/${{ github.repository_owner }}/charts"
+
+      - name: "Generate installation instructions"
+        run: |
+          cat << EOF >> $GITHUB_STEP_SUMMARY
+          ## 📦 Helm Chart Published
+          
+          **Chart:** ${{ env.CHART_NAME }}
+          **Version:** ${{ env.CHART_VERSION }}
+          **Registry:** oci://${{ env.REGISTRY }}/${{ github.repository_owner }}/charts/${{ env.CHART_NAME }}
+          
+          ### Installation
+          
+          \`\`\`bash
+          helm upgrade --install ${{ env.CHART_NAME }} \\
+            oci://${{ env.REGISTRY }}/${{ github.repository_owner }}/charts/${{ env.CHART_NAME }} \\
+            --version ${{ env.CHART_VERSION }} \\
+            -n qgnet --create-namespace
+          \`\`\`
+          
+          ### With values file
+          
+          \`\`\`bash
+          helm upgrade --install ${{ env.CHART_NAME }} \\
+            oci://${{ env.REGISTRY }}/${{ github.repository_owner }}/charts/${{ env.CHART_NAME }} \\
+            --version ${{ env.CHART_VERSION }} \\
+            -f values-dev-cluster-ogdc.yaml \\
```

Comment:
This needs to be pre-processed by `envsubst` first, similarily to how `install-ogdc.sh` script does it.